src/tests/sphmultipcqueue_t.c

```diff
@@ -840,8 +840,14 @@ fill_test_parallel_thread (void *arg)
 		CPU_ZERO_S(size,cset);
 		CPU_SET_S(cpu_list[tn],size,cset);
 		rc = sched_setaffinity(sphFastGetTID(),size,cset);
-		printf("%6d: sched_setaffinity(thread %d to CPU %d)=%d\n",
-		       sphFastGetTID(),tn,cpu_list[tn],rc);
+		if (rc == 0) {
+			printf("%6d: sched_setaffinity(thread %d to CPU %d)\n",
+				sphFastGetTID(),tn,cpu_list[tn]);
+		} else {
+			fprintf(stderr,"%6d: sched_setaffinity(thread %d to CPU %d) ",
+				sphFastGetTID(),tn,cpu_list[tn]);
+			perror(0);
+		}
```

Comment:
I'll add similar changes to sphdirectpcqueue_ttt.
As for the intermingled output, that's an artifact of multiple threads doing output simultaneously. It may be made worse by having the multiple calls to output one line (fprintf; perror). I'll combine them into one call using strerror_r to construct the full line then a single call to fprintf.